src/AspNetCoreModuleV2/CommonLib/requesthandler.h

```diff
@@ -30,7 +38,26 @@ class REQUEST_HANDLER: public virtual IREQUEST_HANDLER
         }
     }
 
-    REQUEST_NOTIFICATION_STATUS OnAsyncCompletion(DWORD cbCompletion, HRESULT hrCompletionStatus) override
+
+    REQUEST_NOTIFICATION_STATUS
+    OnExecuteRequestHandler() final
+    {
+        auto traceScope = TraceContextScope(m_pHttpContext.GetTraceContext());
+        return ExecuteRequestHandler();
+    }
+
+    REQUEST_NOTIFICATION_STATUS
+    OnAsyncCompletion(
+        DWORD      cbCompletion,
+        HRESULT    hrCompletionStatus
+    ) final
+    {
+        auto traceScope = TraceContextScope(m_pHttpContext.GetTraceContext());
```

Comment:
Yeah, if you integrate ModuleTracer and TraceContextScope together, that'd be great.